cardano_node_tests/tests/test_kes.py

```diff
@@ -395,19 +427,33 @@ def test_opcert_future_kes_period(  # noqa: C901
 
         # check kes-period-info with valid operational certificate
         kes_period_info = cluster.get_kes_period_info(valid_opcert_file)
-        kes.check_kes_period_info_result(
-            kes_output=kes_period_info, expected_scenario=kes.KesScenarios.ALL_VALID
+        kes_period_info_errors_list.append(
+            kes.check_kes_period_info_result(
+                kes_output=kes_period_info,
+                expected_scenario=kes.KesScenarios.ALL_VALID,
+                check_id="4",
+            )
         )
 
-        # check kes-period-info with invalid operational certificate, wrong counter and period
+        # check kes-period-info with operational certificate with invalid counter and kes-period
         kes_period_info = cluster.get_kes_period_info(invalid_opcert_file)
-        kes.check_kes_period_info_result(
-            kes_output=kes_period_info,
-            expected_scenario=kes.KesScenarios.INVALID_KES_PERIOD
-            if VERSIONS.cluster_era > VERSIONS.ALONZO
-            else kes.KesScenarios.ALL_INVALID,
+        kes_period_info_errors_list.append(
+            kes.check_kes_period_info_result(
+                kes_output=kes_period_info,
+                expected_scenario=kes.KesScenarios.INVALID_KES_PERIOD
+                if VERSIONS.cluster_era > VERSIONS.ALONZO
+                else kes.KesScenarios.ALL_INVALID,
+                check_id="5",
+            )
         )
 
+        err_joined = "\n".join(e for e in kes_period_info_errors_list if e)
+        if err_joined:
+            if VERSIONS.cluster_era > VERSIONS.ALONZO and "check '2'" in err_joined:
```

Comment:
Proposal:
`if VERSIONS.cluster_era > VERSIONS.ALONZO and len(kes_period_info_errors_list) == 1 and "check '2'" in err_joined:`
ugly, but makes sure we xfail only when there's the single expected error reported.